Seemingly random crashes caused by PegDownProcessor

`Caused by: org.parboiled.errors.ParserRuntimeException: Error while parsing action 'Root/Sequence/ZeroOrMore/Sequence/Block/FirstOf/OrderedList/NodeSequence_Action2'`

Consider switching to https://github.com/vsch/flexmark-java

--- a/app/services/PageProvider.java
+++ b/app/services/PageProvider.java
@@ -1,7 +1,9 @@
 package services;
 
+import com.vladsch.flexmark.html.HtmlRenderer;
+import com.vladsch.flexmark.parser.Parser;
+import com.vladsch.flexmark.util.options.MutableDataSet;
 import org.apache.commons.io.IOUtils;
-import org.pegdown.PegDownProcessor;
 import play.Logger;
 
 import java.io.BufferedReader;
@@ -18,17 +20,21 @@ import java.util.Map;
  */
 public class PageProvider {
 
-  private static PegDownProcessor pegDownProcessor = new PegDownProcessor();
-
   private String mPath;
   private String mExtension;
   private Map<String, Object> mSections;
+  private HtmlRenderer mHtmlRenderer;
+  private Parser mParser;
 
   public PageProvider(String aPath, String aExtension, Map<String, Object> aSections) {
 
     mPath = aPath;
     mExtension = aExtension;
     mSections = aSections;
+
+    MutableDataSet options = new MutableDataSet();
+    mParser = Parser.builder(options).build();
+    mHtmlRenderer = HtmlRenderer.builder(options).build();
 
   }
 
@@ -109,7 +115,7 @@ public class PageProvider {
       line = br.readLine();
     }
 
-    page.put("body", pegDownProcessor.markdownToHtml(IOUtils.toString(br)));
+    page.put("body", mHtmlRenderer.render(mParser.parse(IOUtils.toString(br))));
 
     return page;
 

--- a/build.sbt
+++ b/build.sbt
@@ -26,7 +26,7 @@ libraryDependencies ++= Seq(
   "org.apache.commons" % "commons-email" % "1.3.3",
   "commons-io" % "commons-io" % "2.5",
   "org.elasticsearch" % "elasticsearch" % "2.4.0",
-  "com.vladsch.flexmark" % "flexmark" % "0.19.6",
+  "com.vladsch.flexmark" % "flexmark-all" % "0.26.4",
   "com.github.jknack" % "handlebars" % "4.0.6",
   "com.github.jknack" % "handlebars-markdown" % "4.0.6",
   "org.apache.jena" % "apache-jena-libs" % "3.1.1",

